langchain4j-voyage/src/main/java/dev/langchain4j/model/voyage/VoyageEmbeddingModel.java

```diff
@@ -0,0 +1,255 @@
+package dev.langchain4j.model.voyage;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.DimensionAwareEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.voyage.VoyageApi.DEFAULT_BASE_URL;
+import static dev.langchain4j.model.voyage.VoyageEmbeddingModelName.VOYAGE_3_LITE;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * An implementation of an {@link EmbeddingModel} that uses
+ * <a href="https://docs.voyageai.com/docs/embeddings">Voyage AI Embedding API</a>.
+ */
+public class VoyageEmbeddingModel extends DimensionAwareEmbeddingModel {
+
+    private final VoyageClient client;
+    private final Integer maxRetries;
+    private final VoyageEmbeddingModelName modelName;
+    private final String inputType;
+    private final Boolean truncation;
+    private final String encodingFormat;
+    private final int maxSegmentsPerBatch;
+
+    public VoyageEmbeddingModel(
+            String baseUrl,
+            Duration timeout,
+            Integer maxRetries,
+            String apiKey,
+            VoyageEmbeddingModelName modelName,
+            String inputType,
+            Boolean truncation,
+            String encodingFormat,
+            Boolean logRequests,
+            Boolean logResponses,
+            Integer maxSegmentsPerBatch
+    ) {
+        // Below attributes are force to non-null.
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.modelName = getOrDefault(modelName, VOYAGE_3_LITE);
+        this.truncation = getOrDefault(truncation, true);
+        this.maxSegmentsPerBatch = getOrDefault(maxSegmentsPerBatch, 96);
+        // Below attributes can be null.
+        this.inputType = inputType;
+        this.encodingFormat = encodingFormat;
```

Comment:
Does it make sense to have this configurable if user will receive embeddings as float aray anyway?

Reply:
One scenario I can think of is that the user wants to reduce bandwidth consumption for network transfers when the embedding length is very large and many times. This would be done by setting encodingFormat to base64, and if the user wants to get the results quickly (I'm not sure if base64 compression would increase the response), then encodingFormat could be set to null.

Reply:
Overall, I think this parameter setting does have very little impact on the user, but we should still reserve the right to let the user set it.